packages/next/bin/next

```diff
@@ -16,56 +17,75 @@ if (pkg.peerDependencies) {
 }
 
 const defaultCommand = 'dev'
-const commands = new Set([
+const commands = [
   'init',
   'build',
   'start',
   'export',
   defaultCommand
-])
+]
 
-let cmd = process.argv[2]
-let args = []
-let nodeArgs = []
+const args = arg({
+  // Types
+  '--version': Boolean,
+  '--help': Boolean,
+  '--node-args': String,
+  '--inspect': Boolean,
 
-if (new Set(['--version', '-v']).has(cmd)) {
+  // Aliases
+  '-v': '--version',
+  '-h': '--help'
+}, {
+  // To forward args
+  permissive: true
+})
+
+// Defaults
+const {
+  '--version': version = false,
+  '--help': help = false,
+  '--node-args': nodeArgs = '',
+  '--inspect': inspect = false
+} = args
+
+if (version) {
   console.log(`next.js v${pkg.version}`)
   process.exit(0)
 }
 
-const inspectArg = process.argv.find(arg => arg.includes('--inspect'))
-if (inspectArg) {
-  nodeArgs.push(inspectArg)
-}
-
-if (new Set(['--help', '-h']).has(cmd)) {
+if (help) {
   console.log(`
     Usage
       $ next <command>
 
     Available commands
-      ${Array.from(commands).join(', ')}
+      ${commands.join(', ')}
+      
+    Options
+      --version, -p   Version number
+      --node-args     Node arguments applied to the process
+      --help, -h      Displays this message  
 
     For more information run a command with the --help flag
       $ next init --help
   `)
   process.exit(0)
 }
 
-if (commands.has(cmd)) {
-  args = process.argv.slice(3)
-} else {
-  cmd = defaultCommand
-  args = process.argv.slice(2)
+if (inspect) {
+  nodeArgs.push('--inspect')
 }
 
-const defaultEnv = cmd === 'dev' ? 'development' : 'production'
+const command = args._.find(cmd => commands.includes(cmd)) || defaultCommand
+const forwardedArgs = args._.filter(arg => arg !== command)
+
+const defaultEnv = command === 'dev' ? 'development' : 'production'
 process.env.NODE_ENV = process.env.NODE_ENV || defaultEnv
 
-const bin = join(__dirname, 'next-' + cmd)
+const bin = join(__dirname, 'next-' + command)
 
 const startProcess = () => {
-  const proc = spawn('node', [...nodeArgs, ...[bin], ...args], { stdio: 'inherit', customFds: [0, 1, 2] })
+  const proc = spawn('node', [...nodeArgs.split(' '), bin, ...forwardedArgs], { stdio: 'inherit', customFds: [0, 1, 2] })
```

Comment:
This is one bit of Javascript weirdness lol:

```
''.split('  ')
[ '' ]
```

This is causing an issue here:

```
...nodeArgs.split(' ')
```

Because `''` will be part of the arguments and it screws with `node` spawning:

Eg try running:

```
./node_modules/next/dist/bin/next build ./test/integration/export
```

Reply:
One fix would be just doing:

```
.filter(i => i !== '')
```

Reply:
yea, this will also trim extra spaces